src/legacy/ui/public/saved_objects/components/saved_object_finder.tsx

```diff
@@ -248,73 +328,169 @@ class SavedObjectFinder extends React.Component<SavedObjectFinderProps, SavedObj
             data-test-subj="savedObjectFinderSearchInput"
           />
         </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiFilterGroup>
+            <EuiToolTip position="top" content={this.getCurrentSortTooltip()}>
+              <EuiButtonEmpty
+                size="s"
+                color="text"
+                data-test-subj="savedObjectFinderSortButton"
+                onClick={() =>
+                  this.setState(({ sortDirection, filter }) => ({
+                    sortDirection: this.getNextDirection(sortDirection, Boolean(filter)),
+                  }))
+                }
+              >
+                <FormattedMessage
+                  id="common.ui.savedObjects.finder.sortByLabel"
+                  defaultMessage="Sort: "
+                />
+                <EuiIcon type={this.getCurrentSortIcon()} />
+              </EuiButtonEmpty>
+            </EuiToolTip>
+            {this.props.showFilter && availableSavedObjectMetaData.length > 1 && (
+              <EuiPopover
+                id="popover"
+                panelClassName="euiFilterGroup__popoverPanel"
+                panelPaddingSize="none"
+                isOpen={this.state.filterOpen}
+                closePopover={() => this.setState({ filterOpen: false })}
+                button={
+                  <EuiButtonEmpty
+                    onClick={() =>
+                      this.setState(({ filterOpen }) => ({
+                        filterOpen: !filterOpen,
+                      }))
+                    }
+                    iconSide="right"
+                    iconType="filter"
+                    size="s"
+                    color="text"
+                    data-test-subj="savedObjectFinderFilterButton"
+                  >
+                    <FormattedMessage
+                      id="common.ui.savedObjects.finder.filterButtonLabel"
+                      defaultMessage="Filter"
+                    />
+                  </EuiButtonEmpty>
+                }
+              >
+                <div className="euiFilterSelect__items">
+                  <EuiText className="euiFilterSelectItem savedObjectFilterHeading" size="xs">
+                    <FormattedMessage
+                      id="common.ui.savedObjects.finder.filterLabel"
+                      defaultMessage="Only show the following objects:"
```

Comment:
Is this message needed if the filter button's text is "Types" instead of the generic "Filter"?